api/tests/unit/features/test_unit_features_views.py

```diff
@@ -2510,6 +2510,157 @@ def test_list_features_with_intersection_tag(
     assert response.data["results"][0]["tags"] == [tag1.id, tag2.id]
 
 
+def test_list_features_with_sort_by_state_string_and_int(
+    staff_client: APIClient,
+    project: Project,
+    feature: Feature,
+    with_project_permissions: WithProjectPermissionsCallable,
+    environment: Environment,
+) -> None:
+    # Given
+    with_project_permissions([VIEW_PROJECT])
+    feature2 = Feature.objects.create(
+        name="another_feature", project=project, initial_value="initial_value"
+    )
+    feature3 = Feature.objects.create(
+        name="missing_feature", project=project, initial_value="gone"
+    )
+    feature4 = Feature.objects.create(
+        name="fancy_feature", project=project, initial_value="fancy"
+    )
+
+    Environment.objects.create(
+        name="Out of test scope environment",
+        project=project,
+    )
+
+    feature_state1 = feature.feature_states.filter(environment=environment).first()
+    feature_state1.enabled = True
+    feature_state1.save()
+
+    feature_state_value1 = feature_state1.feature_state_value
+    feature_state_value1.string_value = None
+    feature_state_value1.integer_value = 1945
+    feature_state_value1.save()
+
+    feature_state2 = feature2.feature_states.filter(environment=environment).first()
+    feature_state2.enabled = True
+    feature_state2.save()
+
+    feature_state_value2 = feature_state2.feature_state_value
+    feature_state_value2.string_value = None
+    feature_state_value2.boolean_value = True
```

Comment:
As per my comment above, we might need to also be careful of setting the `type` attribute here.

Reply:
Yeah great catch. I'll fix this and the other test and of course the type check in the view as well.